### PR TITLE
feat: Add vm config parser, unify vm and helm type app deployment

### PR DESCRIPTION
**Pull request by iwisunny**
Refactor app deployment

### CODE REVIEW

File: src/pages/TestDeploy/index.jsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  fetchData = () => new Promise(resolve => setTimeout(resolve, 500));
+
+  componentDidMount() {
+    // mock fetch data
+    this.setState({ loading: true });
+
+    this.fetchData().then(() => {
+      this.parser = new VMParser(mockData);
+
+      this.setState({ loading: false });
```
Comment (whDongRui) [inline src/pages/TestDeploy/index.jsx]: Remove test code.

File: src/pages/TestDeploy/index.jsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      return null;
+    }
+
+    const groups = [].concat(
+      { title: 'Basic settings', items: this.parser.getBasicSetting() },
+      this.parser.getNodeSetting(),
+      { title: 'Vxnet settings', items: this.parser.getVxnetSetting() }
+    );
+
+    console.log('render data: ', groups);
```
[the remark this answers is not in the document]
Comment (iwisunny) [reply src/pages/TestDeploy/index.jsx]: All right, this pr is working in progress, later will remove test code

File: src/stores/app/deploy.js [hunk 3]
```diff
@@ -1,255 +1,132 @@
...
-  async fetchSubnets(runtimeId) {
-    const result = await this.request.get(`clusters/subnets`, {
+  fetchSubnetsByRuntime = async runtimeId => {
+    this.isLoading = true;
+    const result = await this.request.get(`clusters/subnets`, { 
       runtime_id: runtimeId,
       limit: this.maxLimit
     });
     this.subnets = get(result, 'subnet_set', []);
-    this.subnetId = this.subnets[0] ? this.subnets[0].subnet_id : '';
```
Comment (whDongRui) [inline src/stores/app/deploy.js]: No need remove this code. If selected subnet, then change selected runtime, the subnet will show previous value.

File: src/stores/app/deploy.js [hunk 4]
```diff
@@ -1,255 +1,132 @@
...
+
+    const files = get(result, 'files', {});
+
+    if (files['config.json']) {
+      this.configJson = JSON.parse(Base64.decode(files['config.json']));
+    } else if (files['values.yaml']) {
+      this.yamlStr = Base64.decode(files['values.yaml']);
     } else {
-      this.yamlConfig = [];
+      this.errMsg = 'Invalid config file, failed to render page';
```
Comment (whDongRui) [inline src/stores/app/deploy.js]: Add relatived translation.

Comment (iwisunny) [reply src/stores/app/deploy.js]: OK

File: src/stores/app/deploy.js (hunk 3 above)
Comment (iwisunny) [reply src/stores/app/deploy.js]: No, if changed runtime, newly subnets will be fetched, previous subnet won't be set.

Comment (whDongRui) [reply src/stores/app/deploy.js]: Previous has a bug, after change runtime, query subnets no data, but the page subnet  label show previous value.

**Pull request merged**